src/nvim/ex_docmd.c

```diff
@@ -577,6 +577,14 @@ int do_cmdline(char_u *cmdline, LineGetter fgetline,
         ++RedrawingDisabled;
         did_inc = TRUE;
       }
+      if (State & CMDPREVIEW) {
```

Comment:
I could also add the condition in the above if stmt to check for `CMDPREVIEW`, and remove the DOCMD_NOWAIT from the part where this function will be called, for the case of previewable

Reply:
If there is need to shorten the code that would be good

Reply:
changing the above would be better. why did you removed `++no_wait_return` ?